From 99a8070b5863ac6aae19d86b5342d04b457f009f Mon Sep 17 00:00:00 2001
From: Yongjun Zhang <yzhang@cloudera.com>
Date: Fri, 11 Dec 2015 18:44:01 -0800
Subject: [PATCH 1038/2748] HDFS-9519. Some coding improvement in
 SecondaryNameNode#main. (Xiao Chen via Yongjun
 Zhang)

(cherry picked from commit 2a4c7d4facabb8b99d6dcbf4ccfe2afedf4fd445)
(cherry picked from commit 3fc9593318364c9b24f1116b570c56f3f751db26)

Change-Id: Icf587c685a04fafdd73159bf25eab2b05dcfe1e1
---
 .../hdfs/server/namenode/SecondaryNameNode.java    |   13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/SecondaryNameNode.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/SecondaryNameNode.java
index 8605bbd..e1485d4 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/SecondaryNameNode.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/SecondaryNameNode.java
@@ -675,16 +675,15 @@ public static void main(String[] argv) throws Exception {
       SecondaryNameNode secondary = null;
       secondary = new SecondaryNameNode(tconf, opts);
 
+      // SecondaryNameNode can be started in 2 modes:
+      // 1. run a command (i.e. checkpoint or geteditsize) then terminate
+      // 2. run as a daemon when {@link #parseArgs} yields no commands
       if (opts != null && opts.getCommand() != null) {
+        // mode 1
         int ret = secondary.processStartupCommand(opts);
         terminate(ret);
-      }
-
-      if (secondary != null) {
-        // The web server is only needed when starting SNN as a daemon,
-        // and not needed if called from shell command. Starting the web server
-        // from shell may fail when getting credentials, if the environment
-        // is not set up for it, which is most of the case.
+      } else {
+        // mode 2
         secondary.startInfoServer();
         secondary.startCheckpointThread();
         secondary.join();
-- 
1.7.9.5

